packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -280,8 +280,11 @@ export function TableNG(props: TableNGProps) {
 
     // Helper function to get displayed value
     const getDisplayedValue = (row: TableRow, key: string) => {
-      const field = props.data.fields.find((field) => field.name === key)!;
-      const displayedValue = formattedValueToString(field.display!(row[key]));
+      const field = props.data.fields.find((field) => getDisplayName(field) === key);
+      if (!field || !field.display) {
+        return '';
```

Comment:
the empty string is just a bit of a smell to me here, especially given that this is then used as the key in a set. My biggest concern is with situations where two fields fall into this edge case, or what if the displayName is actually `''`. I think it's a little easier to reason about if this method returns null in this case, and then the usage of `getDisplayedValue` has to handle null.